crates/duckdb/src/client.rs

```diff
@@ -186,16 +190,28 @@ impl Client {
     /// let item_collection = client.search("data/100-sentinel-2-items.parquet", Default::default()).unwrap();
     /// ```
     pub fn search(&self, href: &str, search: Search) -> Result<stac_api::ItemCollection> {
-        let record_batches = self.search_to_arrow(href, search)?;
-        if record_batches.is_empty() {
-            Ok(Default::default())
-        } else {
-            let schema = record_batches[0].schema();
-            let item_collection = stac::geoarrow::json::from_record_batch_reader(
-                RecordBatchIterator::new(record_batches.into_iter().map(Ok), schema),
-            )?;
-            Ok(item_collection.into())
-        }
+        let mut arrow_iter = self.search_to_arrow(href, search)?;
+        let Some(schema) = arrow_iter.schema() else {
+            return Ok(Default::default());
+        };
+
+        let first_batch = match arrow_iter.next() {
+            Some(batch) => batch?,
+            None => return Ok(Default::default()),
+        };
+
+        let batches = std::iter::once(Ok(first_batch))
+            .chain(arrow_iter)
+            .map(|batch| {
+                batch.map_err(|err| {
+                    ArrowError::ExternalError(Box::new(ArrowExternalError(err.to_string())))
+                })
+            });
```

Comment:
I think this can just be:


```suggestion
            .map(|batch| batch.map_err(|err| ArrowError::ExternalError(Box::new(err))));
```